cmd/agent/api/internal/agent/agent.go

```diff
@@ -501,14 +501,13 @@ func getDiagnose(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Serizalize it
-	diagnosesJson, err := json.Marshal(diagnoses)
+	err = json.NewEncoder(w).Encode(diagnoses)
```

Comment:
Should this happen after `Header().Set()`? I'm assuming `Encode` will call `w.Write`, which would render changes to headers ineffective (see second paragraph [here](https://pkg.go.dev/net/http#ResponseWriter)).

Reply:
I was pondering about it. It works as it but I think it is better to move it per your suggestion at least to keep the same call flow pattern. Thank you (will be pushing in a few minutes).